server/src/main/java/org/opensearch/action/search/SearchRequestStats.java

```diff
@@ -75,6 +89,23 @@ protected void onPhaseFailure(SearchPhaseContext context, Throwable cause) {
         phaseStatsMap.get(context.getCurrentPhase().getSearchPhaseName()).current.dec();
     }
 
+    @Override
+    protected void onRequestStart(SearchRequestContext searchRequestContext) {
+        tookStatsHolder.current.inc();
+    }
+
+    @Override
+    protected void onRequestEnd(SearchPhaseContext context, SearchRequestContext searchRequestContext) {
+        tookStatsHolder.current.dec();
+        tookStatsHolder.total.inc();
+        tookStatsHolder.timing.inc(TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - searchRequestContext.getAbsoluteStartNanos()));
+    }
+
+    @Override
+    protected void onRequestFailure(SearchPhaseContext context, SearchRequestContext searchRequestContext) {
+        tookStatsHolder.current.dec();
+    }
```

Comment:
Should we track the time on failure too? (This isn't an opinion, just curious what the reasoning is here)

Reply:
Thanks @jed326, that is a good point. This follows the same logic as existing coordinator phase stats. Maybe failure times can skew the average or this is the metric definition? I will note this and discuss as a possible improvement.

Reply:
Yep I guess it will depend on what you think this metric will be mostly used for.